constr.mm

```diff
@@ -0,0 +1,139 @@
+    fldextdgirred.e $e |- ( ph -> E e. ( SubDRing ` F ) ) $.
+    fldextdgirred.p $e |- ( ph -> P e. ( Base ` M ) ) $.
+    fldextdgirred.1 $e |- ( ph -> P e. ( Irred ` M ) ) $.
+    fldextdgirred.z $e |- ( ph -> Z e. ( B \ E ) ) $.
+    fldextdgirred.2 $e |- ( ph -> ( ( O ` P ) ` Z ) = .0. ) $.
+    $( If ` Z ` is the root of an irreducible polynomial ` P ` over a field
+       ` K ` , then the degree of the field extension ` [ K [ Z ] : K ] ` is
+       the degree of the polynomial ` P ` . $)
+    fldextdgirred $p |- ( ph -> ( L [:] K ) = ( D ` P ) ) $= ? $.
+  $}
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Quadratic field extensions
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c /FldExt2 $.
+
+  $( Extend class notation with the quadratic field extension relation. $)
+  cfldext2 $a class /FldExt2 $.
+
+  ${
+    $d e f $.
+    $( Define the quadratic field extension relation.  Quadratic field
+       extensions are field extensions of degree 2.  $)
+    df-fldext2 $a |- /FldExt2 = { <. e , f >. | ( e /FldExt f /\ ( e [:] f ) = 2 ) } $.
+  $}
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Chain
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c Chain $.
+
+  $( Extend class notation with the (finite) chain builder function. $)
+  cchn $a class Chain $.
+
+  ${
+    $d r c n $.
+    $( Define the (finite) chain builder function.  A chain is defined to be a
+      sequence of objects, whereas each object is in a given relation with the
+      next one. $)
+    df-chn $a |- Chain = ( r e. _V |-> { c e. Word dom r |
+      A. n e. ( dom c \ { 0 } ) ( c ` ( n - 1 ) ) r ( c ` n ) } ) $.
+  $}
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Constructible points
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c Constr $.
+
+  $( Extend class notation with the set of geometrically constructible points. $)
+  cconstr $a class Constr $.
+
+  $( Define the set of geometrically constructible points, by recursively adding
+     the line-line, line-circle and circle-circle intersections constructions
+     using points in a previous iteration. $)
+  df-constr $a |- Constr = ( rec ( ( s e. _V |-> { x e. CC |
+    ( E. a e. s E. b e. s E. c e. s E. d e. s E. t e. RR E. r e. RR ( x = ( a + ( t x. ( b - a ) ) ) /\ x = ( c + ( r x. ( d - c ) ) ) /\ -. ( Im ` ( ( * ` ( b - a ) ) x. ( d - c ) ) ) = 0 )
+     \/ E. a e. s E. b e. s E. c e. s E. e e. s E. f e. s E. t e. RR ( x = ( a + ( t x. ( b - a ) ) ) /\ ( abs ` ( x - c ) ) = ( abs ` ( e - f ) ) )
+     \/ E. a e. s E. b e. s E. c e. s E. d e. s E. e e. s E. f e. s E. t e. RR ( -. a = d /\ ( abs ` ( x - a ) ) = ( abs ` ( b - c ) ) /\ ( abs ` ( x - d ) ) = ( abs ` ( e - f ) ) ) )
+      } ) , { 0 , 1 } ) " _om ) $.
```

Comment:
Very long and complicated! Would it be helpful to have separate definitions for "set of intersections of two lines",   "set of intersections of a line and a circle" and "set of intersections of two circles"?

Reply:
We try to limit the number of definitions - do you think it would be wise to add them? I don't know about any other use.

Do you think we could reuse any of the theorems of the [`~ inlinecirc02p `](https://us.metamath.org/mpeuni/inlinecirc02p.html) family?

Reply:
> We try to limit the number of definitions - do you think it would be wise to add them? I don't know about any other use.

OK, maybe we can provide theorems instead which show the three kinds of construction explicitly.

> 
> BTW, do you think we could reuse any of the theorems of the [`~ inlinecirc02p `](https://us.metamath.org/mpeuni/inlinecirc02p.html) family?

Maybe, but the definitions are too different at the moment. Maybe we need a mapping of the different representations first.

Reply:
> OK, maybe we can provide theorems instead which show the three kinds of construction explicitly.

I have suggested some theorems in this [post](https://github.com/metamath/set.mm/issues/4246#issuecomment-2590323747) which decompose the definition into three subgoals. We don't have the algebraic definition of constructible numbers with towers yet, so they can't be used as is, but maybe we could restate those only in terms of fields. For example, for a line, given by two points, and a circle, given by three points, if these points lie in some subfield of C, then the intersection also lies in quadratic extension of that subfield.

I think we unfortunately can't use `inlinecirc02p` since we need to keep track of the subfield we are working in.